src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantOverlapAnnotator.java

```diff
@@ -0,0 +1,204 @@
+    }
+
+    /**
+     * Add overlaps flag attributes to vcToAnnotate binding overlapTestVCs.getSource() => true if
+     * an overlapping variant context can be found in overlapTestVCs with vcToAnnotate
+     *
+     * Overlaps here means that the reference alleles are the same and at least one alt
+     * allele in vcToAnnotate is equals to one of the alt alleles in overlapTestVCs
+     *
+     * @param overlapTestVCs a non-null list of potential overlaps that start at vcToAnnotate
+     * @param attributeKey the key to set to true in the attribute map for vcToAnnotate if it overlaps
+     * @param vcToAnnotate a non-null VariantContext to annotate
+     * @return
+     */
+    public VariantContext annotateOverlap(final List<VariantContext> overlapTestVCs, final String attributeKey, final VariantContext vcToAnnotate) {
+        if ( overlapBindings.isEmpty() ) {
+            return vcToAnnotate;
+        }
+
+        final boolean overlaps = overlaps(overlapTestVCs, vcToAnnotate);
+        if ( overlaps ) {
+            return new VariantContextBuilder(vcToAnnotate).attribute(attributeKey, true).make();
+        } else {
+            return vcToAnnotate;
+        }
+    }
+
+    /**
+     * Returns the ID field of the first VariantContext in rsIDSourceVCs that has the same reference allele
+     * as vcToAnnotate and all of the alternative alleles in vcToAnnotate.
+     *
+     * Doesn't require vcToAnnotate to be a complete match, so
+     *
+     * A/C/G in VC in rsIDSourceVCs
+     *
+     * would match the a VC with A/C but not A/T.  Also we don't require all alleles to match
+     * so we would also match A/C/T to A/C/G.
+     *
+     * Will only match rsIDSourceVCs that aren't failing filters.
+     *
+     * @param rsIDSourceVCs a non-null list of potential overlaps that start at vcToAnnotate
+     * @param vcToAnnotate a non-null VariantContext to annotate
+     * @return a String to use for the rsID from rsIDSourceVCs if one matches, or null if none matches
+     */
+    private static String getRsID(final List<VariantContext> rsIDSourceVCs, final VariantContext vcToAnnotate) {
+        Utils.nonNull(rsIDSourceVCs, "rsIDSourceVCs cannot be null");
+        Utils.nonNull(vcToAnnotate, "vcToAnnotate cannot be null");
+
+        for ( final VariantContext vcComp : rsIDSourceVCs ) {
+            if ( vcComp.isFiltered() ) {
+                continue; // don't process any failed VCs
+            }
+
+            if ( ! vcComp.getContig().equals(vcToAnnotate.getContig()) || vcComp.getStart() != vcToAnnotate.getStart() ) {
+                throw new IllegalArgumentException("source rsID VariantContext " + vcComp + " doesn't start at the same position as vcToAnnotate " + vcToAnnotate);
+            }
+
+            if ( vcToAnnotate.getReference().equals(vcComp.getReference()) ) {
+                for ( final Allele allele : vcToAnnotate.getAlternateAlleles() ) {
+                    if ( vcComp.getAlternateAlleles().contains(allele) ) {
```

Comment:
Good to know that this checks that the alleles match -- I think not all of our dbSNP comparisons do.